app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by(email: params[:email])
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
```

Comment:
maybe put this in a new method? `user && !user.activated? && user.authenticated?(:activation, params[:id])`